Sideset generator refactor

pbehne opened the pull request:
Clean up sideset generation and factor out commonalities in a base class.

Closes #26252.

pbehne committed:
Moved some common variables to SideSetsGeneratorBase, changed increament loops to range-based loops.

pbehne committed:
General cleanup.

Factored some parameters/variables from derived SideSetsGeneratorBase classes
into parent. Combined duplicate _variance and _normal_tol parameters into a single
_normal_tol parameter and consolidated duplicate normal_tol comparisons into a single
method in the base class.

GiudGiud commented on framework/include/meshgenerators/SideSetsGeneratorBase.h:
```diff
@@ -52,13 +52,42 @@ class SideSetsGeneratorBase : public MeshGenerator
+  bool addSideToBoundary(const Elem * elem, const unsigned int side) const;
+
+  /// the mesh to add the sidesets to
+  std::unique_ptr<MeshBase> & _input;
+
+  const bool _fixed_normal;
 
   /// Whether or not to remove the old sidesets (if any) when adding sidesets
-  bool _replace;
+  const bool _replace;
+
+  /// Whether to only include external side when considering sides to add to the sideset
+  const bool _include_only_external_sides;
+
+  /// Whether to skip side if it is already part of existing sideset
+  const bool _skip_if_part_of_existing_sideset;
+
+  /// Whether to only include if side is already part of existing sideset
+  const bool _include_only_if_part_of_existing_sideset;
+
+  ///// Whether to only include if side is touching a subdomain
+  // const bool _include_only_if_touching_subdomain;
+
+  /// if specified, then faces are only added if their normal is close to this
+  Point _normal;
+
+  /// true if only faces close to "normal" will be added
+  bool _using_normal;
```
yup this is all great consolidation

pbehne replied:
Should we be concerned that not every one of these variables applies to every derived class? For instance, `_include_only_external_sides` is used in `ParsedGenerateSideset` and `SideSetsAroundSubdomainGenerator`, but not in `AllSideSetsByNormalsGenerator`, `SideSetsFromNormalsGenerator`, or `SideSetsFromPointsGenerator`. As currently implemented, the later three always only include external sides, so there is no need to check the value of `_include_only_external_sides` in these generators. In some cases, it could make sense to incorporate `_include_only_external_sides` into these generators. For instance, in `AllSideSetsByNormalsGenerator`, if `_include_only_external_sides` is set to false and a subdomain id is specified, the result would look very similar to something from `SideSetsAroundSubdomainGenerator`. Do we want to make these kind of modifications, or just be okay with some derived classes not using some of the base class parameters?

GiudGiud replied:
it s ok to have derived class remove parameters.
set them to the decent value then use suprressParam

GiudGiud replied:
We should also consider if we have any other sideset-generating class that can inherit from this base class and gain capability. Maybe not

GiudGiud replied:
> Do we want to make these kind of modifications

if we can gain meaningful capability at little efforts I would say yes. But usually it s best to be needs-driven. As you are generating the mesh you realize you want this, and because it s in the base class it's very clean to add it

pbehne replied:
> We should also consider if we have any other sideset-generating class that can inherit from this base class and gain capability. Maybe not

Yeah, the base class could also be applicable to the following generators. I'll check before I'm ready to close.

SideSetsFromNodeSetsGenerator
SideSetsFromBoundingBoxGenerator
RefineSidesetGenerator
SideSetsBetweenSubdomainsGenerator

GiudGiud commented on framework/src/meshgenerators/ParsedGenerateSideset.C:
```diff
@@ -75,23 +66,19 @@ ParsedGenerateSideset::validParams()
-    _input(getMesh("input")),
     _function(parameters.get<std::string>("combinatorial_geometry")),
     _sideset_name(getParam<BoundaryName>("new_sideset_name")),
     _check_boundaries(isParamValid("included_boundaries")),
     _check_subdomains(isParamValid("included_subdomain_ids") ||
                       isParamValid("included_subdomains")),
     _check_neighbor_subdomains(isParamValid("included_neighbor_ids") ||
                                isParamValid("included_neighbors")),
```
these 3 could be moved upwards too

GiudGiud commented on framework/src/meshgenerators/SideSetsGeneratorBase.C:
```diff
@@ -36,14 +35,52 @@ SideSetsGeneratorBase::validParams()
+                                  "The variance allowed when comparing normals",
+                                  "Deprecated, use 'normal_tol' instead");
+  params.deprecateParam("variance", "normal_tol", "4/01/2024");
+
```
I d add a parameter group, something like "sideset restriction" (parameters)

GiudGiud reviewed (commented):
yeah this is exactly the kind of clean up I was thinking about
awesome

pbehne committed:
Cleaup.

Moved more parameters into base class suppressed unused parameters in inherited
classes, created sideset restriction parameter group. Ref #26252.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2081513/) on b3ed4d8 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/26869/site)

This comment will be updated on new commits.

pbehne committed:
Removed deprecated parameters.

pbehne commented:
@GiudGiud, ready for review. I look at the other sideset generators that do not currently inherit from base, and it does not seem to me that they are similar enough to base to gain anything by inheriting from it.

GiudGiud commented:
I would not "closes" for the ticket then. But this is already a big step forward
I would expect these 3
```
SideSetsFromNodeSetsGenerator
SideSetsFromBoundingBoxGenerator
SideSetsBetweenSubdomainsGenerator
```
can be factored in to have the extra options

pbehne committed:
Fixed more deprecation.

pbehne committed:
Removed more param deprecation stuff, supporessed more params.

pbehne commented:
Alright, I added `SideSetsFromNodeSetsGenerator`, `SideSetsFromBoundingBoxGenerator`, and `SideSetsBetweenSubdomainsGenerator` to inherit from the base class. If all tests pass, hopefully we are good!

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2082591/) on c0ef994 wanted to post the following:


The following coverage requirement(s) failed:
- Failed to generate `framework` coverage rate (required: 81.0%)
- Failed to generate `chemical_reactions` coverage rate (required: 92.0%)
- Failed to generate `electromagnetics` coverage rate (required: 94.0%)
- Failed to generate `external_petsc_solver` coverage rate (required: 85.0%)
- Failed to generate `fluid_properties` coverage rate (required: 84.0%)
- Failed to generate `fsi` coverage rate (required: 85.0%)
- Failed to generate `functional_expansion_tools` coverage rate (required: 81.0%)
- Failed to generate `heat_transfer` coverage rate (required: 87.0%)
- Failed to generate `misc` coverage rate (required: 32.0%)
- Failed to generate `navier_stokes` coverage rate (required: 77.0%)
- Failed to generate `optimization` coverage rate (required: 86.0%)
- Failed to generate `ray_tracing` coverage rate (required: 93.0%)
- Failed to generate `scalar_transport` coverage rate (required: 85.0%)
- Failed to generate `solid_properties` coverage rate (required: 83.0%)
- Failed to generate `thermal_hydraulics` coverage rate (required: 89.0%)

pbehne commented:
> Job [Coverage](https://civet.inl.gov/job/2082591/) on [c0ef994](https://github.com/idaholab/moose/commit/c0ef9944be7b7c63aa65580e63259a58c19071f1) wanted to post the following:
> 
> The following coverage requirement(s) failed:
> 
>     * Failed to generate `framework` coverage rate (required: 81.0%)
> 
>     * Failed to generate `chemical_reactions` coverage rate (required: 92.0%)
> 
>     * Failed to generate `electromagnetics` coverage rate (required: 94.0%)
> 
>     * Failed to generate `external_petsc_solver` coverage rate (required: 85.0%)
> 
>     * Failed to generate `fluid_properties` coverage rate (required: 84.0%)
> 
>     * Failed to generate `fsi` coverage rate (required: 85.0%)
> 
>     * Failed to generate `functional_expansion_tools` coverage rate (required: 81.0%)
> 
>     * Failed to generate `heat_transfer` coverage rate (required: 87.0%)
> 
>     * Failed to generate `misc` coverage rate (required: 32.0%)
> 
>     * Failed to generate `navier_stokes` coverage rate (required: 77.0%)
> 
>     * Failed to generate `optimization` coverage rate (required: 86.0%)
> 
>     * Failed to generate `ray_tracing` coverage rate (required: 93.0%)
> 
>     * Failed to generate `scalar_transport` coverage rate (required: 85.0%)
> 
>     * Failed to generate `solid_properties` coverage rate (required: 83.0%)
> 
>     * Failed to generate `thermal_hydraulics` coverage rate (required: 89.0%)

@GiudGiud, what exactly does this imply? To my knowledge, coverage measures how much of the code the test suite actually tests. I didn't expect to see massive failures like this. Would a rebase to devel or next help?

GiudGiud commented:
This just means the coverage recipe crapped out.Did you rebase this PR on next by any chance? There are no coverage measurements for most versions of next (because we dont let it finish for example). Only devel is reliably measured for coverage

pbehne commented:
I've only rebased on devel, though not recently. My understanding is that testing is done on a framework created by merging the PR branch with devel, so rebasing typically isn't necessary. Should I rebase to devel now that the tensor mechanics stuff has been fixed?

GiudGiud commented:
Rebasing on devel is one way to reset this

GiudGiud commented on framework/include/meshgenerators/SideSetsGeneratorBase.h:
```diff
@@ -52,13 +52,46 @@ class SideSetsGeneratorBase : public MeshGenerator
   void flood(const Elem * elem, Point normal, boundary_id_type side_id, MeshBase & mesh);
 
-  boundary_id_type getNextBoundaryID() const;
+  bool normalsWithinTol(const Point & normal_1, const Point & normal_2) const;
```
```suggestion
  bool normals_within_tol(const Point & normal_1, const Point & normal_2) const;
```
camel case is for class names

GiudGiud replied:
add docstring too

GiudGiud replied:
oh you re right

GiudGiud commented on framework/include/meshgenerators/SideSetsGeneratorBase.h:
```diff
@@ -52,13 +52,46 @@ class SideSetsGeneratorBase : public MeshGenerator
+  /// the mesh to add the sidesets to
+  std::unique_ptr<MeshBase> & _input;
+
+  const bool _fixed_normal;
```
docstrings

GiudGiud commented on framework/include/meshgenerators/SideSetsGeneratorBase.h:
```diff
@@ -52,13 +52,46 @@ class SideSetsGeneratorBase : public MeshGenerator
+
+  const bool _fixed_normal;
 
   /// Whether or not to remove the old sidesets (if any) when adding sidesets
```
```suggestion
  /// Whether or not to remove the old sidesets (all of them, if any) when adding sidesets
```

GiudGiud commented on framework/include/meshgenerators/SideSetsGeneratorBase.h:
```diff
@@ -52,13 +52,46 @@ class SideSetsGeneratorBase : public MeshGenerator
+  /// whether to check boundary ids when adding sides or not
+  const bool _check_boundaries;
+
+  /// whether to check subdomain ids when adding sides or not
```
```suggestion
  /// whether to check subdomain ids of the element in the (element, side, boundary id) tuple when adding sides or not
```

GiudGiud commented on framework/include/meshgenerators/SideSetsGeneratorBase.h:
```diff
@@ -52,13 +52,46 @@ class SideSetsGeneratorBase : public MeshGenerator
+  /// whether to check subdomain ids when adding sides or not
+  const bool _check_subdomains;
+
+  /// whether to check neighbor subdomain ids when adding sides or not
```
```suggestion
  /// whether to check the subdomain ids of the neighbor element (on the other 'side' of the side) when adding sides or not
```

GiudGiud commented on framework/src/meshgenerators/SideSetsGeneratorBase.C:
```diff
@@ -104,13 +147,13 @@ SideSetsGeneratorBase::flood(const Elem * elem,
     {
       if (_replace)
         mesh.get_boundary_info().remove_side(elem, side);
 
```
I would expect the current  elem and neighbor subdomain element id checks to happen here?

GiudGiud replied:
same for check boundaries

GiudGiud commented on framework/src/meshgenerators/SideSetsFromNodeSetsGenerator.C:
```diff
@@ -16,16 +16,23 @@ registerMooseObject("MooseApp", SideSetsFromNodeSetsGenerator);
-  InputParameters params = MeshGenerator::validParams();
+  InputParameters params = SideSetsGeneratorBase::validParams();
+
+  params.suppressParameter<Point>("normal");
+  params.suppressParameter<Real>("normal_tol");
+  params.suppressParameter<bool>("fixed_normal");
+  params.suppressParameter<bool>("replace");
+  params.suppressParameter<bool>("include_only_external_sides");
+  params.suppressParameter<std::vector<BoundaryName>>("included_boundaries");
+  params.suppressParameter<std::vector<SubdomainName>>("included_subdomains");
+  params.suppressParameter<std::vector<SubdomainName>>("included_neighbors");
```
yeah this is not good. You were right initially, does not make sense for this one in its current implementation

GiudGiud replied:
I dont see us implementing the rest of the parameters

GiudGiud replied:
so let's roll back on this one

GiudGiud commented on framework/src/meshgenerators/SideSetsFromBoundingBoxGenerator.C:
```diff
@@ -59,13 +58,21 @@ SideSetsFromBoundingBoxGenerator::validParams()
   params.addParam<MooseEnum>(
       "location", location, "Control of where the subdomain id is to be set");
 
+  params.suppressParameter<Point>("normal");
```
```suggestion
  // TODO: Implement each of these in the generate() routine using utilities in SidesetGeneratorBase
  params.suppressParameter<Point>("normal");
```

GiudGiud commented on framework/src/meshgenerators/SideSetsBetweenSubdomainsGenerator.C:
```diff
@@ -20,38 +20,39 @@ registerMooseObject("MooseApp", SideSetsBetweenSubdomainsGenerator);
   params.addClassDescription("MeshGenerator that creates a sideset composed of the nodes located "
                              "between two or more subdomains.");
 
+  params.suppressParameter<Point>("normal");
```
same add a note to the developer

GiudGiud commented on framework/src/meshgenerators/AllSideSetsByNormalsGenerator.C:
```diff
@@ -30,14 +30,19 @@ AllSideSetsByNormalsGenerator::validParams()
-  params.addRequiredParam<MeshGeneratorName>("input", "The mesh we want to modify");
+  params.suppressParameter<Point>("normal");
+  params.suppressParameter<Real>("normal_tol");
+  params.suppressParameter<bool>("include_only_external_sides");
+  params.suppressParameter<std::vector<BoundaryName>>("included_boundaries");
+  params.suppressParameter<std::vector<SubdomainName>>("included_subdomains");
+  params.suppressParameter<std::vector<SubdomainName>>("included_neighbors");
```
these 4, if implemented in flood() would simply work

pbehne committed:
Addressed Guillaume's review.

pbehne committed:
Consolidated common code between SideSetsGeneratorBase::flood and ParsedGenerateSideset::generate.

pbehne committed:
More cleanup, consolidation, deprecation.

pbehne commented:
@GiudGiud, thanks for the comments, they helped me realize other things I could clean up!

moosebuild commented on large_media:
```diff
@@ -1 +1 @@
-Subproject commit f7f7a383fc9eb79b2c1702beac9914423cb01d8d
+Subproject commit 199a23c71d556343cbac89ceffbc987428aa4464
```
**Caution!** This contains a submodule update

pbehne replied:
No? Do I need to rebase to devel?

GiudGiud replied:
haha yes it does include. civet dont lie

rebase on devel will solve this, you ll get a conflict and just dont use your submodule hash for large_media

GiudGiud commented on large_media:
see

pbehne committed:
Unscrewing large_media screw up (hopefully)

moosebuild commented on large_media:
```diff
@@ -1 +1 @@
-Subproject commit f7f7a383fc9eb79b2c1702beac9914423cb01d8d
+Subproject commit 199a23c71d556343cbac89ceffbc987428aa4464
```
**Caution!** This contains a submodule update

GiudGiud replied:
still there. you ll need to rebase with `e` as in edit for the commit changing that.
then use `git reset HEAD~1 large_media` to reset large media. then add it and finish the rebase

pbehne replied:
The change to `large_media` is no longer showing up in the `File Changed` section and the original warning is now outdated. Are you sure the issue is still there?

GiudGiud replied:
yup seems gone now

pbehne commented:
I only ran the tests in the `meshgenerators` folder locally. I'll dig into the failing tests now and see what I can find.

pbehne commented:
> I only ran the tests in the `meshgenerators` folder locally. I'll dig into the failing tests now and see what I can find.

See #13346.

pbehne committed:
Fixed minor bugs, updated param names for tests using --error-deprecated

pbehne committed:
Added include for non-unity build.

pbehne committed:
Modified some includes for non-unity build.

pbehne committed:
Fixed bug with default tolerance value.

pbehne committed:
Updated deprecated param names in documentation.

pbehne committed:
Removed suppression of 'fixed_normal' in SideSetsFromNormalsGenerator

pbehne committed:
Added clarification in comment.

pbehne commented:
@GiudGiud, this is ready for another review. Everything is passing except for Griffin documentation because it references the name of a deprecated parameter. We can fix this if you agree with the changes and nothing gets changed/reverted.

GiudGiud commented:
A deprecated parameter? Link should still work. did you remove it?
or is an old page name?

pbehne commented on framework/src/meshgenerators/SideSetsBetweenSubdomainsGenerator.C:
```diff
@@ -20,67 +20,44 @@ registerMooseObject("MooseApp", SideSetsBetweenSubdomainsGenerator);
-      "The 'master_block' param is deprecated and will be removed on January 1, 2021. Please use "
-      "the 'primary_block' parameter instead.");
-  params.addRequiredParam<std::vector<SubdomainName>>(
-      "paired_block", "The paired set of blocks for which to draw a sideset between");
-  params.addRequiredParam<std::vector<BoundaryName>>("new_boundary",
-                                                     "The name of the boundary to create");
+      "Deprecated, use 'included_subdomains'");
+  params.deprecateParam("primary_block", "included_subdomains", "4/01/25");
```
Deprecation of `primary_block` happens here. See civet job https://civet.inl.gov/job/2115751/

pbehne replied:
Would you mind reviewing everything so far? That way, I don't have to switch between tasks as much.

GiudGiud commented on framework/src/meshgenerators/SideSetsAroundSubdomainGenerator.C:
```diff
@@ -25,22 +25,14 @@ SideSetsAroundSubdomainGenerator::validParams()
+      "block",
+      "The blocks around which to create sidesets",
+      "Deprecated, use 'included_subdomains' instead");
+  params.deprecateParam("block", "included_subdomains", "4/01/25");
```
same let s try to rename. if it works, better than deprecation

pbehne commented on framework/include/meshgenerators/SideSetsGeneratorBase.h:
```diff
@@ -50,15 +48,87 @@ class SideSetsGeneratorBase : public MeshGenerator
+   * included_boundaries parameter.
+   */
+  bool elementSideInIncludedBoundaries(const Elem * elem,
+                                       const uint & side,
```
Replacing with unsigned int since we haven't heard from Alex.

GiudGiud replied:
sounds good

pbehne commented on framework/include/meshgenerators/SideSetsGeneratorBase.h:
```diff
@@ -50,15 +48,87 @@ class SideSetsGeneratorBase : public MeshGenerator
+  /// whether to check the subdomain ids of the neighbor element (on the other 'side' of the side) when adding sides
+  const bool _check_neighbor_subdomains;
+
+  /// A list of included boundary ids that the side has to be part of
+  std::vector<boundary_id_type> _restricted_boundary_ids;
+
+  /// A list of included subdomain ids that the side has to be part of
+  std::vector<subdomain_id_type> _included_subdomain_ids;
```
`_restricted_boundary_ids` are extracted from the `included_boundaries` parameter, while `_included_subdomain_ids` are extracted from the `included_subdomains` parameter. I'm adding documentation that states this clarification.

pbehne committed:
Addressed Guillaume's review.

pbehne committed:
Undoing parameter renames in documentation.

pbehne commented:
> seems there's some doco items to update. Let s talk about the way that does not involve renaming too much stuff, I dont want you to lose time on renaming if we can avoid it

This is because I previously changed a .md file when renaming primary and paired blocks. I just undid the commit. Hopefully that resolves it.

GiudGiud commented on framework/include/meshgenerators/SideSetsAroundSubdomainGenerator.h:
```diff
@@ -28,24 +28,8 @@ class SideSetsAroundSubdomainGenerator : public SideSetsGeneratorBase
-
-  /// if specified, then faces are only added if their normal is close to this
-  Point _normal;
+  static bool elemSideOnBoundary(const Elem * elem, const uint & side);
```
why static?
```suggestion
  static bool elemSideOnBoundary(const Elem * const elem, const unsigned int side);
```

pbehne replied:
I made it static because it does not rely on data from a particular `SideSetsAroundSubdomainGenerator` object. Would it make sense to instead move it out of the class into a utility function? I could also remove the static and make the method const, but I don't see why removing the static is necessary or desirable because the method doesn't need access to object data.

pbehne replied:
`elemSideOnBoundary` is used twice in `generate`, but not anywhere else. Would it make sense to define it as a lambda function in `generate`?

GiudGiud replied:
so is this routine more of a elemSideOnExternalBoundary ?
I dont see how we can check if it s on a particular boundary in the restricted set without using object data

pbehne replied:
According to my docstring, the method determines "whether the given side of an element resides on an external or internal boundary"
Here is the implementation. It does not require the use of object data because the element and side in question are passed as arguments and the relevant neighbor element is extracted from those parameters.
https://github.com/idaholab/moose/blob/95a82d112c0259fe2ac54bab612c4abf6fb5ed7f/framework/src/meshgenerators/SideSetsAroundSubdomainGenerator.C#L186-L191

pbehne replied:
We could also rename the method to `elemSideOnInternalOrExternalBoundary` if you think it would add clarity.

pbehne replied:
We could also move this method into the base class.

GiudGiud commented on framework/include/meshgenerators/SideSetsGeneratorBase.h:
```diff
@@ -50,15 +48,87 @@ class SideSetsGeneratorBase : public MeshGenerator
+   * @param tol The comparison tolerance.
+   * @return A bool indicating whether 1 - dot(normal_1, normal_2) <= tol.
+   */
+  static bool normalsWithinTol(const Point & normal_1, const Point & normal_2, const Real & tol);
```
why static
```suggestion
  static bool normalsWithinTol(const Point & normal_1, const Point & normal_2, const Real tol);
```

pbehne replied:
Same argument as above.

GiudGiud commented on framework/include/meshgenerators/SideSetsGeneratorBase.h:
```diff
@@ -50,15 +48,87 @@ class SideSetsGeneratorBase : public MeshGenerator
+   * include_only_external_sides, included_boundaries, included_neighbor_subdomains and normal
+   */
+  bool elemSideSatisfiesRequirements(const Elem * const elem,
+                                     const unsigned int & side,
```
```suggestion
                                     const unsigned int side,
```

GiudGiud commented on framework/include/meshgenerators/SideSetsGeneratorBase.h:
```diff
@@ -50,15 +48,87 @@ class SideSetsGeneratorBase : public MeshGenerator
+  /// A list of included subdomain ids that the side has to be part of, extracted from the included_subdomains parameter
+  std::vector<subdomain_id_type> _included_subdomain_ids;
+
+  /// A list of included neighbor subdomain ids
```
```suggestion
  /// A list of included neighbor subdomain ids that the sides' neighbor element must be a part of
```

GiudGiud commented on framework/include/meshgenerators/SideSetsGeneratorBase.h:
```diff
@@ -50,15 +48,87 @@ class SideSetsGeneratorBase : public MeshGenerator
+  const bool _include_only_external_sides;
+
+  /// if specified, then faces are only added if their normal is close to this
+  Point _normal;
```
```suggestion
  const Point _normal;
```

GiudGiud commented on framework/include/meshgenerators/SideSetsGeneratorBase.h:
```diff
@@ -50,15 +48,87 @@ class SideSetsGeneratorBase : public MeshGenerator
+  Point _normal;
+
+  /// true if only faces close to "normal" will be added
+  bool _using_normal;
```
```suggestion
  const bool _using_normal;
```

GiudGiud commented on framework/include/meshgenerators/SideSetsGeneratorBase.h:
```diff
@@ -50,15 +48,87 @@ class SideSetsGeneratorBase : public MeshGenerator
+   * if face_normal.normal_hat <= 1 - normal_tol
+   * where normal_hat = _normal/|_normal|
+   */
+  Real _normal_tol;
```
```suggestion
  const Real _normal_tol;
```

GiudGiud commented on framework/src/meshgenerators/AllSideSetsByNormalsGenerator.C:
```diff
@@ -30,14 +30,20 @@ AllSideSetsByNormalsGenerator::validParams()
   InputParameters params = SideSetsGeneratorBase::validParams();
 
-  params.addRequiredParam<MeshGeneratorName>("input", "The mesh we want to modify");
+  params.setParameters("include_only_external_sides", true);
```
```suggestion
  // This is the expected behavior of this sideset generator
  params.setParameters("include_only_external_sides", true);
```

GiudGiud commented on framework/src/meshgenerators/AllSideSetsByNormalsGenerator.C:
```diff
@@ -30,14 +30,20 @@ AllSideSetsByNormalsGenerator::validParams()
+  params.setParameters("include_only_external_sides", true);
+  params.suppressParameter<bool>("include_only_external_sides");
+
+  params.suppressParameter<bool>("fixed_normal");
```
```suggestion
  // The normals are found from the actual orientation of sidesets, not user-specified
  params.suppressParameter<bool>("fixed_normal");
```

GiudGiud commented on framework/src/meshgenerators/ParsedGenerateSideset.C:
```diff
@@ -27,95 +27,34 @@ ParsedGenerateSideset::validParams()
       {},
       "Vector of values for the constants in constant_names (can be an FParser expression)");
+
+  params.suppressParameter<std::vector<BoundaryName>>("new_boundary");
```
```suggestion
  // This sideset generator can only handle a single new sideset name
  params.suppressParameter<std::vector<BoundaryName>>("new_boundary");
```

GiudGiud commented on framework/src/meshgenerators/SideSetsGeneratorBase.C:
```diff
@@ -87,37 +185,99 @@ SideSetsGeneratorBase::finalize()
+  // Skip if element does not have neighbor in specified subdomains
+  if (_check_neighbor_subdomains)
+  {
+    const Elem * neighbor = elem->neighbor_ptr(side);
```
```suggestion
    const Elem * const neighbor = elem->neighbor_ptr(side);
```
or const auto

GiudGiud commented on framework/src/meshgenerators/SideSetsGeneratorBase.C:
```diff
@@ -87,37 +185,99 @@ SideSetsGeneratorBase::finalize()
+      return false;
+  }
+
+  // We'll just use the normal of the first qp
```
not sure about that comment.
where was the first qp decision made?

pbehne committed:
Addressed Guillaume review.

pbehne committed:
Changed static methods to const.

GiudGiud commented on framework/src/meshgenerators/SideSetsAroundSubdomainGenerator.C:
```diff
@@ -230,3 +181,11 @@ SideSetsAroundSubdomainGenerator::generate()
+SideSetsAroundSubdomainGenerator::elemSideOnBoundary(const Elem * const elem,
+                                                     const unsigned int side) const
+{
+  const Elem * neighbor = elem->neighbor_ptr(side);
+  return (neighbor == nullptr) or (elem->subdomain_id() != neighbor->subdomain_id());
```
```suggestion
  const auto neighbor = elem->neighbor_ptr(side);
  return (neighbor == nullptr) || (elem->subdomain_id() != neighbor->subdomain_id());
```

GiudGiud commented on framework/src/meshgenerators/SideSetsFromPointsGenerator.C:
```diff
@@ -131,7 +133,7 @@ SideSetsFromPointsGenerator::generate()
   finalize();
 
-  for (unsigned int i = 0; i < boundary_ids.size(); ++i)
+  for (const auto i : make_range(boundary_ids.size()))
```
```suggestion
  for (const auto i : index_range(boundary_ids))
```

GiudGiud commented on framework/src/meshgenerators/SideSetsFromPointsGenerator.C:
```diff
@@ -74,7 +76,7 @@ SideSetsFromPointsGenerator::generate()
   std::unique_ptr<PointLocatorBase> pl = PointLocatorBase::build(TREE, *mesh);
 
-  for (unsigned int i = 0; i < boundary_ids.size(); ++i)
+  for (const auto i : make_range(boundary_ids.size()))
```
```suggestion
  for (const auto i : index_range(boundary_ids))
```

pbehne committed:
Apply suggestions from code review

Co-authored-by: Guillaume Giudicelli <guillaume.giudicelli@gmail.com>

GiudGiud reviewed (approved):
looks great. Thanks for taking care of this.
